plugins/in_opentelemetry/opentelemetry_prot.c

```diff
@@ -2435,19 +2436,37 @@ int opentelemetry_prot_handle_ng(struct flb_http_request *request,
         strcmp(request->path, "/opentelemetry.proto.collector.metric.v1.MetricService/Export") == 0 ||
         strcmp(request->path, "/opentelemetry.proto.collector.metrics.v1.MetricsService/Export") == 0) {
         payload_type = 'M';
-        result = process_payload_metrics_ng(context, context->ins->tag, request, response);
+        if (context->tag_from_uri == FLB_TRUE) {
+            tag = flb_sds_create("v1_metrics");
```

Comment:
This should be collided when the same in_opentelemetry are instantiated cases.
How about adding `ins->name` as a suffix?

Reply:
This would be a breaking change with the existing behaviour already defined within the HTTP/1.1 code path:

https://github.com/fluent/fluent-bit/blob/13f96f940ec6fd8097ed5f978e4f6e816e24a970/plugins/in_opentelemetry/opentelemetry_prot.c#L1831-L1853

I agree that your proposed behaviour probably makes more sense but might make sense to do it as a separate PR across both codepaths. At the moment you can't get *any* tag set at all in the (default) http2 codepath.

Longer-term it might make more sense to have the tag_from_uri feature work on an append or replace basis but this would require more code changes

Reply:
Hmm.., I agree with your opinion. BTW, any chance to unify the behavior of HTTP/1.1 one?
This is because the following code:

```c
/* New tag skipping the URI '/' */ 
     flb_sds_cat(tag, uri + 1, len - 1); 
  
     /* Sanitize, only allow alphanum chars */ 
     for (i = 0; i < flb_sds_len(tag); i++) { 
         if (!isalnum(tag[i]) && tag[i] != '_' && tag[i] != '.') { 
             tag[i] = '_'; 
         } 
```

could be used in the HTTP/2 code path, too. What do you think?

Reply:
It seemed pretty unnecessary to me to copy this over... the URIs are already string checked a lot here:

https://github.com/fluent/fluent-bit/blob/13f96f940ec6fd8097ed5f978e4f6e816e24a970/plugins/in_opentelemetry/opentelemetry_prot.c#L2434-L2447

...and around the point of my patch too.

Yes, the same sanitize code could be copy-pasted, but in the gRPC cases this would lead to a different (and quite unwieldly IMO) tag of `opentelemetry_proto_collector_metrics_v1_MetricsService/Export` (in the metrics case, as well as the other equivalents for logs and traces). This would also lead to different tags for data that came in over gRPC v.s. HTTP.

For routing purposes I think it's probably a lot more likely that people will want to direct all traces to one place (e.g. jaeger), all metrics to another place (prometheus?), and all logs to potentially some other third place, regardless of whether they came in over HTTP or gRPC. Which is why I left it with the same v1_xxx strings hardcoded for now.

As per my earlier comment [here](https://github.com/fluent/fluent-bit/issues/8734#issuecomment-2134178486) on #8734, I think what would make more sense long term is a `tag_from_signal` option that allows replacing or appending just `metric`/`trace`/`log` to the tag based on the telemetry signal type and not the URI. If people really needed tag-based routing for gRPC v.s. HTTP this could be handled with another `tag_from_protocol` option or something like that.

I started some work on a `tag_from_signal` option the other week with append support but it was going to take more time and I just wanted to get this bug fixed somehow first.

Reply:
Got it. Then, this PR should be reasonable for me.

Reply:
:raised_hands: thanks Hiroshi!